smart-contracts/contracts/Unlock.sol

```diff
@@ -315,6 +317,37 @@ contract Unlock is UnlockInitializable, UnlockOwnable {
     return newLock;
   }
 
+  /**
+   * Create an upgradeable lock using a specific PublicLock version, and execute
+   * transaction(s) on the created lock.
+   * @param data bytes containing the call to initialize the lock template
+   * (refer to createUpgradeableLock for more details).
+   * Importantly, the initial lock manager needs to be set to this contract.
+   * @param lockVersion the version of the lock to use
+   * @param transactions an array of transactions to be executed on the newly
+   * created lock. It is recommended to include a transaction to renounce the
+   * lock manager as the last transaction.
+   */
+  function createUpgradeableLockAtVersion(
+    bytes memory data,
+    uint16 lockVersion,
+    bytes[] calldata transactions
+  ) public returns (address newLock) {
+    newLock = createUpgradeableLockAtVersion(data, lockVersion);
+    // Execute all transactions
+    for (uint256 i = 0; i < transactions.length; i++) {
+      (bool success, ) = newLock.call(transactions[i]);
+      if (!success) {
+        revert(
+          string.concat(
+            "LockProxyDeployer: transaction failed",
```

Comment:
Having a literal string may add to the size of the contract significantly. Better to use a custom error

Reply:
also both OZ and 3rd web have a way to detect forwarded calls. Maybe we can just use one of these lib  

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Multicall.sol

https://github.com/thirdweb-dev/contracts/blob/main/contracts/extension/Multicall.sol

Reply:
Good point! Let me do that then!

Reply:
ok so when looking at the OZ  and Thirdweb libs they seem to focus on delegate calls, where the code executed is in the context of the caller (here that would be Unlock), which is not really what I would want to achieve...